source/run-command/tests/unified/run-command.yml

```
@@ -0,0 +1,242 @@
+              commandName: ping
+
+  - description: attaches the provided session lsid to given command
+    operations:
+      - name: runCommand
+        object: *plainDb
+        arguments:
+          commandName: ping
+          command:
+            ping: 1
+          session: *session
+        expectResult:
+          ok: 1
+    expectEvents:
+      - client: *client
+        events:
+        - commandStartedEvent:
+              command:
+                ping: 1
+                lsid: { $$sessionLsid: *session }
+                $db: *plainDb
+              commandName: ping
+
+  - description: attaches the provided $readPreference to given command
+    operations:
+      - name: runCommand
+        object: *plainDb
+        arguments:
+          commandName: ping
+          command:
+            ping: 1
+          readPreference: 'nearest'
+        expectResult:
+          ok: 1
+    expectEvents:
+      - client: *client
+        events:
+        - commandStartedEvent:
+              command:
+                ping: 1
+                $readPreference: { mode: 'nearest' }
+                $db: *plainDb
+              commandName: ping
+
+  - description: does not inherit readConcern specified at the db level
+    operations:
+      - name: runCommand
+        object: *dbWithRC
+        arguments:
+          commandName: ping
+          command:
+            ping: 1
+        expectResult:
+          ok: 1
+    expectEvents:
+      - client: *client
+        events:
+        - commandStartedEvent:
+              command:
+                ping: 1
+                readConcern: { $$exists: false }
+                $db: *dbWithRC
+              commandName: ping
+
+  - description: does not inherit writeConcern specified at the db level
+    operations:
+      - name: runCommand
+        object: *dbWithWC
+        arguments:
+          commandName: ping
+          command:
+            ping: 1
+        expectResult:
+          ok: 1
+    expectEvents:
+      - client: *client
+        events:
+        - commandStartedEvent:
+              command:
+                ping: 1
+                writeConcern: { $$exists: false }
+                $db: *dbWithWC
+              commandName: ping
+
+  - description: does not retry retryable errors on given command
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ping]
+              closeConnection: true
+      - name: runCommand
+        object: *plainDb
+        arguments:
+          commandName: ping
+          command:
+            ping: 1
+        expectError:
+          isError: true
+
+  - description: attaches transaction fields to given command
+    runOnRequirements:
+      - topologies: [ replicaset, sharded-replicaset, load-balanced, sharded ]
+        minServerVersion: '4.4'
```

```suggestion
      - minServerVersion: "4.0"
        topologies: [ replicaset ]
      - minServerVersion: "4.2"
        topologies: [ sharded-replicaset, load-balanced ]
```

Not sure why `minServerVersion: 4.4` was used, but other transaction tests allow 4.0+ replica sets and 4.2+ sharded clusters.

Specifying `sharded-replicaset` and `sharded` together also seemed redundant.

I don't think `sharded` should be used for any spec tests that require successfully executing a transaction since an oplog is needed. It looks like there are a few instances of tests requiring `sharded` instead of `sharded-replicaset`. I'll open a separate JIRA ticket about that.

---

Just to close the loop here, I created [DRIVERS-2609](https://jira.mongodb.org/browse/DRIVERS-2609) to clear up other spec tests.